packages/common/src/util.js

```diff
@@ -723,7 +728,11 @@ function promisify(originalMethod) {
           return reject(err);
         }
 
-        resolve(callbackArgs);
+        if (options.singular && callbackArgs.length === 1) {
```

Comment:
I think we also need a test for when they specified `singular`, but `callbackArgs.length` is > 1.